src/platform/Tizen/DiagnosticDataProviderImpl.cpp

```diff
@@ -34,6 +189,475 @@ DiagnosticDataProviderImpl & DiagnosticDataProviderImpl::GetDefaultInstance()
     return sInstance;
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapFree(uint64_t & currentHeapFree)
+{
+    struct mallinfo mallocInfo = mallinfo();
+    currentHeapFree            = mallocInfo.fordblks;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapUsed(uint64_t & currentHeapUsed)
+{
+    struct mallinfo mallocInfo = mallinfo();
+    currentHeapUsed            = mallocInfo.uordblks;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapHighWatermark(uint64_t & currentHeapHighWatermark)
+{
+    struct mallinfo mallocInfo = mallinfo();
+    currentHeapHighWatermark   = mallocInfo.uordblks;
```

Comment:
This is the same as GetCurrentHeapUsed.  That does not seem right.

Reply:
It would be better to modify it to unimplemented. i'll update it.